### DISCUSSION TITLE
connection closed before message completed

### DISCUSSION THREAD

**Discussion by josejachuf** [salvo-rs/salvo#307] [Q&A] [answered]
I am developed an app to serve vector tiles.

The flow is like that:

[request (tiles//z/x/y.pbf)] -> 

```
[handler] -> [fn getTile] -> [if tile in disk {
                                open tile
                                return tile
                              }else {
                                create tile (database)
                                save tile on disk
                                return tile
                              }
                             ] -> [handler response tile]       
```
The disk cache is asynchronous using tokyo::fs

On some occasions I get this error

> ERROR salvo_core::server: http serve connection failed error=Custom { kind: Other, error: "connection closed before message completed" }

I know that it is not much explanation that I give, but can it be due to the disk reading?

Tiles are used to visualize maps. In a panning or zoom the requests to the server are many. Can this be?

Thanks 
Jose

**Comment by chrislearn** [comment, marked as answer]
I removed it in main branch.

**Comment by josejachuf** [comment]
So, don't I worry about the error?
